include/logging/log_msg2.h

```diff
@@ -388,8 +395,11 @@ do { \
 			   _level, _data, _dlen, \
 			   FOR_EACH_IDX(Z_LOG_LOCAL_ARG_NAME, (,), __VA_ARGS__)); \
 } while (0)
-#endif /* CONFIG_LOG2_ALWAYS_RUNTIME */
+#endif /* CONFIG_LOG2_ALWAYS_RUNTIME || LOG_DISABLED_OLD_TOOLCHAIN_WORKAROUND */
 
+#ifdef LOG_DISABLED_OLD_TOOLCHAIN_WORKAROUND
+#undef LOG_DISABLED_OLD_TOOLCHAIN_WORKAROUND
+#endif
```

Comment:
Why undef it?  Not understanding this part.  This header has an include guard anyway, so it's not like you're going to have variant/duplicate definitions of this floating around a compilation unit.

Reply:
In general, just like to clean up things that should not be used after this.